src/kernels/gpu_reference_kernel/naive_conv.cpp

```diff
@@ -1025,6 +1036,132 @@ inline __device__ void naive_conv_fwd_ndhwc(const src_data_t* __restrict__ p_in,
         p_out[o_idx] = cast_to<acc_data_t, dst_data_t>(value);
     }
 }
+
+// design block_size 256
+template <typename src_data_t, typename acc_data_t, typename dst_data_t>
+inline __device__ void naive_conv_fwd_ndhwc_nonpacked(const src_data_t* __restrict__ p_in,
+                                            const src_data_t* __restrict__ p_wei,
+                                            dst_data_t* __restrict__ p_out,
+                                            Strides5D in_strides,
+                                            Strides6D wei_strides,
+                                            Strides5D out_strides,
+                                            int di,
+                                            int hi,
+                                            int wi,
+                                            int n,
+                                            int k_per_group,
+                                            int c_per_group,
```

Comment:
Shall we make some of the parameters unsigned (or even explicit `uint32_t`). At least we'll get twice bigger range.

Another suggestion is to make some of the parameters as `uint64_t` to be able to process big tensors. The same suggestion applies to `Strides*D` types.

I guess most of the parameters or even all of them are unsigned, but I didn't dig deep and probably some of them can be negative.

Reply:
Will consider it as a follow up work. Jehandad and I discussed 64-bit parameters and decided against it.

Reply:
@JehandadKhan @amberhassaan The rule of thumb is that any parameter that includes result of multiplication of 4 or more dimensions should be 64-bit. See the implementation of naive forward pooling solver.
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/d176b8ff5cfcfe7c01a8af210f145d57a98ab3f4/src/solver/pooling/forwardNaive.cpp#L122-L133

Reply:
@amberhassaan @CAHEK7 please mask this as NOT resolved,

Reply:
@atamazov @CAHEK7 : I have switched to 64-bit integers for strides. Please review the PR again. Thank you.